Orange/widgets/data/oraclesql.py

```diff
@@ -0,0 +1,122 @@
+    priority = 1
+    autocommit = settings.Setting(False, schema_only=True)
+    savedQuery = settings.Setting(None, schema_only=True)
+    savedUsername = settings.Setting(None, schema_only=True)
+    savedPwd = settings.Setting(None, schema_only=True)
+    savedDB = settings.Setting(None, schema_only=True)
+
+    def __init__(self):
+        super().__init__()
+
+        #Defaults
+        self.setSizePolicy(QSizePolicy.Maximum, QSizePolicy.Maximum)
+        self.domain = None
+        self.data = None
+        self.query = ''
+        if self.savedQuery is not None:
+            self.query = self.savedQuery
+        self.username = ''
+        if self.savedUsername is not None:
+            self.username = self.savedUsername
+        self.password = ''
+        if self.savedPwd is not None:
+            self.password = self.savedPwd
+        self.database = ''
+        if self.savedDB is not None:
+            self.database = self.savedDB
+        #Control Area layout
+        sip.delete(self.controlArea.layout())
+        self.controlArea.setLayout(QHBoxLayout())
+        self.connectBox = gui.widgetBox(self.controlArea, "Database connection")
+        self.connectBox.setSizePolicy(QSizePolicy.Fixed, QSizePolicy.Fixed)
+        self.sqlBox = gui.widgetBox(self.controlArea, "SQL")
+        #Database
+        self.userLabel = gui.label(self.connectBox, self, 'User name')
+        self.connectUser = QLineEdit(self.username, self)
+        self.connectBox.layout().addWidget(self.connectUser)
+        self.passwordLabel = gui.label(self.connectBox, self, 'Password')
+        self.connectPassword = QLineEdit(self.password, self)
+        self.connectPassword.setEchoMode(QLineEdit.Password)
+        self.connectBox.layout().addWidget(self.connectPassword)
+        self.dbLabel = gui.label(self.connectBox, self, 'Database')
+        self.connectDB = QLineEdit(self.database, self)
+        self.connectBox.layout().addWidget(self.connectDB)
+        self.runSQL = gui.auto_commit(self.connectBox, self, 'autocommit',
+                                      label='Run SQL', commit=self.commit)
+        # query
+        self.queryTextEdit = QPlainTextEdit(self.query, self)
+        self.sqlBox.layout().addWidget(self.queryTextEdit)
+        if self.autocommit:
+            self.commit()
+
+    def handleNewSignals(self):
+        self._invalidate()
+
+    def commit(self):
+        username = self.connectUser.text()
+        password = self.connectPassword.text()
+        database = self.connectDB.text()
+        con = cx_Oracle.connect(username+"/"+password+"@"+database)
+        query = self.queryTextEdit.toPlainText()
+        cur = con.cursor()
+        cur.execute(query)
+        results = cur.fetchall()
+        columns = [i[0] for i in cur.description]
+        df = pandas.DataFrame(results, columns=columns)
+        self.send("Dataframe", df)
+        orangetable = self.df2table(df)
+        #orangedomain = self.df2domain(df)
```

Comment:
Don't leave in commented code unless you add a comment about it, why it's still there etc.